### PR TITLE
Use ISO codes available at boundary nodes in ENTSO-E extensions

### PR DESCRIPTION
**Pull request by zamarrenolm**
This change uses the country ISO codes available at boundary nodes (Connectivity or Topological nodes) in ENTSO-E extensions for CGMES when the CGMES import is configured to convert boundaries to the IIDM model.

Currently the country set in the Substations created in IIDM for the boundary nodes is a default value.

With this change, the Country for Substations related to the boundary will be set based on attributes of the CGMES nodes. The attributes `fromEndIsoCode` / `toEndIsoCode` are used to exchange the ISO code of the region to which the "From" /  "To" side of the Boundary point belongs to or it is connected to. Not enough information is present in the model to determine which "side" should be used, so the country will be assigned from the first code that can be mapped to a valid IIDM Country enum value.

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CountryConversion.java [hunk 1]
```diff
@@ -71,7 +70,23 @@ private CountryConversion() {
...
+        }
+        return Optional.empty();
+    }
+
     public static Country defaultCountry() {
         return Country.values()[0];
     }
+
+    public static Country defaultCountryForBoundary() {
+        return Country.BD;
```
Comment (mathbagu) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CountryConversion.java]: Could you add a log to indicate that this method is called? I think it’s important to know that we invent information

**Comment by MioRtia** [comment]
> > According to the documentation, `fromEndIsoCode` and `toEndIsoCode` are mandatory attributes of `ConnectivityNode`. Why are they considered as optional in the sparkl request? Are there cases where they don't appear? (On the other hand, I agree that a default boundary Country should be defined as there is no explicit insurance that these attributes match our `Country` object...)
> 
> These attributes are mandatory for boundary nodes. The OPTIONAL clause in the SPARQL is required because it is obtaining all Connectivity Nodes in the model (not only connectivity nodes on the boundary).

Ah yes, you are right. 
After reflection, I wonder if `defaultBoundaryCountry()` is really relevant though? We can simply throw an exception since `fromEndIsoCode` and `toEndIsoCode` are mandatory for boundaries and there is a finished defined list of ISO codes for countries? If no country corresponds, this is probably a faulty file. What do you think?

**Comment by zamarrenolm** [comment]
> > > According to the documentation, `fromEndIsoCode` and `toEndIsoCode` are mandatory attributes of `ConnectivityNode`. Why are they considered as optional in the sparkl request? Are there cases where they don't appear? (On the other hand, I agree that a default boundary Country should be defined as there is no explicit insurance that these attributes match our `Country` object...)
> > 
> > 
> > These attributes are mandatory for boundary nodes. The OPTIONAL clause in the SPARQL is required because it is obtaining all Connectivity Nodes in the model (not only connectivity nodes on the boundary).
> 
> Ah yes, you are right.
> After reflection, I wonder if `defaultBoundaryCountry()` is really relevant though? We can simply throw an exception since `fromEndIsoCode` and `toEndIsoCode` are mandatory for boundaries and there is a finished defined list of ISO codes for countries? If no country corresponds, this is probably a faulty file. What do you think?

Default boundary country was used to avoid stopping the conversion just because of not finding a valid country. Country is required in IIDM but it does not seem a critical data to be able to perform power system analysis, so we try to fix the missing data with the default value to allow the user to complete the import of the model.

**Comment by zamarrenolm** [comment]
We have decided to throw an Exception if none of the countries interpreted from ISO code are valid as IIDM countries.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CountryConversion.java [hunk 2]
```diff
@@ -71,7 +75,32 @@ private CountryConversion() {
...
+            LOG.warn("Default country converting {}", c.what());
+        }
+        return defaultCountry();
+    }
+
     public static Country defaultCountry() {
         return Country.values()[0];
     }
+
+    public static Country defaultCountryForBoundary() {
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CountryConversion.java]: You can delete this method and then, it is okay for me.

Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CountryConversion.java]: Done

**Comment by MioRtia** [comment]
Your last commit was about `BusValidation` and did not delete `defaultCountryForBoundary`, is that normal?

**Comment by zamarrenolm** [comment]
> Your last commit was about `BusValidation` and did not delete `defaultCountryForBoundary`, is that normal?

That was an error. Sorry for that. I will fix it now

**Pull request merged**